src/v/ssx/thread_worker.h

```diff
@@ -186,10 +267,11 @@ class thread_worker {
  * class is most suited to run for the lifetime of an application, rather than
  * short-lived.
  */
-class thread_worker {
+class singleton_thread_worker {
 public:
     static constexpr ss::shard_id shard_id = 0;
-    thread_worker() = default;
+    explicit singleton_thread_worker(thread_worker_config config)
+      : _config(std::move(config)){};
```

Comment:
Should the shard_id above come from the config now?

Reply:
No - it should probably always be from the thread that `start` is called (which is what it is now), while I like making things explicitly I think it'd have to be a different type for the sharded form.

But now that you say that, the sharded form should probably always pin to a core so the threads don't clump up on a small subset of the cores so I should probably split the configs now.